cmd/webhook/main.go

```diff
@@ -0,0 +1,74 @@
+package main
+
+import (
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/net/context"
+	"gopkg.in/alecthomas/kingpin.v2"
+
+	"github.com/zalando/skipper/cmd/webhook/admission"
+)
+
+const (
+	defaultAddress = ":8080"
+)
+
+func main() {
+	var (
+		certFile string
+		keyFile  string
+		address  string
+	)
+
+	kingpin.Flag("tls-cert-file", "File containing the certificate for HTTPS").Required().Envar("CERT_FILE").StringVar(&certFile)
+	kingpin.Flag("tls-key-file", "File containing the private key for HTTPS").Required().Envar("KEY_FILE").StringVar(&keyFile)
+	kingpin.Flag("address", "The address to listen on").Default(defaultAddress).StringVar(&address)
+
+	kingpin.Parse()
+
+	rgAdmitter := admission.RouteGroupAdmitter{}
+	handler := http.NewServeMux()
+	handler.Handle("/routegroups", admission.Handler(rgAdmitter))
+	handler.Handle("/metrics", promhttp.Handler())
+	handler.HandleFunc("/healthz", healthCheck)
+
+	// One can use generate_cert.go in https://golang.org/pkg/crypto/tls
+	// to generate cert.pem and key.pem.
+	serve(address, certFile, keyFile, handler)
+}
+
+func healthCheck(writer http.ResponseWriter, _ *http.Request) {
+	writer.WriteHeader(http.StatusOK)
+	if _, err := writer.Write([]byte("ok")); err != nil {
+		log.Errorf("failed to write health check: %v", err)
+	}
+
+}
+
+func serve(address, certFile, keyFile string, handler http.Handler) {
+	server := &http.Server{
+		Addr:    address,
+		Handler: handler,
+	}
+
+	log.Infof("Starting server on %s", address)
+
+	sig := make(chan os.Signal, 1)
+	signal.Notify(sig, syscall.SIGTERM)
+	go func() {
+		<-sig
+		server.Shutdown(context.Background())
+	}()
+
+	err := server.ListenAndServeTLS(certFile, keyFile)
```

Comment:
I think we should support non-HTTPS, too, at least for local testing. I think the logic should be something like this:

- if only one of TLS cert or key is provided => error
- if both TLS cert and key are provided, and no address is provided => listenTLS on 9443
- if both TLS cert and key are provided, and address is provided, too => listenTLS on the provided address
- if no TLS cert is provided, and no address is provided => listenPlain on 9080
- if no TLS cert is provided, and address is provided => listenPlain on the provided address